common/lib/xmodule/xmodule/modulestore/split_mongo/split.py

```diff
@@ -1202,9 +1203,14 @@ def _block_matches_all(block_data):
             block_name = qualifiers.pop('name')
             block_ids = []
             for block_id, block in course.structure['blocks'].iteritems():
-                # Do an in comparison on the name qualifier
-                # so that a list can be used to filter on block_id
-                if block_id.id in block_name and _block_matches_all(block):
+                # Don't do an in comparison blindly; first check to make sure
+                # that the name qualifier we're looking at isn't a plain string;
+                # if it is a string, then it should match exactly.
+                if isinstance(block_name, six.string_types):
+                    name_matches = block_id.id == block_name
+                else:
+                    name_matches = block_id.id in block_name
```

Comment:
Please leave in the comment explaining that this line is "so that a list can be used to filter on block_id"